exonum-java-binding/core/rust/ejb-app/src/node_builder.rs

```diff
@@ -17,66 +17,78 @@
 use exonum_btc_anchoring::ServiceFactory as BtcAnchoringServiceFactory;
 use exonum_configuration::ServiceFactory as ConfigurationServiceFactory;
 use exonum_time::TimeServiceFactory;
-use java_bindings::exonum::helpers::fabric::{self, ServiceFactory};
-use java_bindings::utils::{
-    load_enabled_services, BTC_ANCHORING_SERVICE, CONFIGURATION_SERVICE, EJB_SERVICE,
-    PATH_TO_SERVICES_TO_ENABLE, TIME_SERVICE,
+use java_bindings::{
+    exonum::helpers::fabric::{self, ServiceFactory},
+    utils::services::{
+        load_services_definition, system_service_names::*, EjbAppServices,
+        PATH_TO_SERVICES_DEFINITION,
+    },
+    JavaServiceFactoryAdapter,
 };
-use java_bindings::JavaServiceFactory;
-
-use std::collections::{HashMap, HashSet};
-use std::path::Path;
-
-fn service_factories() -> HashMap<String, Box<ServiceFactory>> {
-    let mut service_factories = HashMap::new();
-    service_factories.insert(
-        CONFIGURATION_SERVICE.to_owned(),
-        Box::new(ConfigurationServiceFactory) as Box<ServiceFactory>,
-    );
-    service_factories.insert(
-        BTC_ANCHORING_SERVICE.to_owned(),
-        Box::new(BtcAnchoringServiceFactory) as Box<ServiceFactory>,
-    );
-    service_factories.insert(
-        TIME_SERVICE.to_owned(),
-        Box::new(TimeServiceFactory) as Box<ServiceFactory>,
-    );
-    service_factories.insert(
-        EJB_SERVICE.to_owned(),
-        Box::new(JavaServiceFactory) as Box<ServiceFactory>,
-    );
-    service_factories
+
+use std::{
+    collections::{HashMap, HashSet},
+    path::Path,
+};
+
+/// Creates `NodeBuilder` using services configuration from `ejb_app_services.toml` located in the working directory.
+pub fn create() -> fabric::NodeBuilder {
+    let service_factories = prepare_service_factories(PATH_TO_SERVICES_DEFINITION);
+    let mut builder = fabric::NodeBuilder::new();
+    for (_, service_factory) in service_factories {
```

Comment:
Why this private method returns a HashMap (and is more complicated) if it is not needed?

Reply:
Re-written back to Vector.